Codec/Archive/Tar/Pack.hs

```diff
@@ -74,20 +78,38 @@ pack
   -> IO [Entry]
 pack = packAndCheck (const Nothing)
 
+-- | 'pack'' is like 'pack', but doesn't read the contents of files,
+-- only creating entries with 'OsPath' as contents.
+--
+-- @since 0.7.0.0
+pack'
+  :: FilePath
+  -> [FilePath]
+  -> IO [GenEntry OsPath TarPath LinkTarget]
```

Comment:
Given that we are going for a major version bump anyway, maybe it's time to switch from `FilePath` to `OsPath` everywhere... Although one can argue that `GenEntry OsPath` is a detail implementation and should be semi-hidden behind a type synonym...

(Not asking you to change anything, just thinking aloud)